src/bit.rs

```diff
@@ -1,169 +0,0 @@
-/// Yields the bits of a `u64`.
-pub struct BitIterator {
-    /// The value that is being iterated bit-wise
-    value: u64,
-    /// True bits in the `mask` are the bits that have been visited.
-    mask: u64,
-}
-
-impl BitIterator {
```

Comment:
question: this is basically an iterator API for the `NodeIndex`, so what I got from this is that we don't want to have an iterator for it?

I think that having an iterator is easier to understand that have state mutation on the loop body.

Reply:
I think for now it is simpler without it. The iterator was like 100+ lines of code which, in my opinion, made what's going on non-obvious. I replaced it with something like 4 lines of code which I think are not too difficult to interpret.

We can of course come back to it later on if we can use it more widely.